sympy/simplify/cse_main.py

```diff
@@ -491,10 +491,10 @@ def tree_cse(exprs, symbols, opt_subs=None, order='canonical', ignore=()):
 
     ## Find repeated sub-expressions
 
-    to_eliminate = set()
+    to_eliminate = OrderedSet()
 
-    seen_subexp = set()
-    excluded_symbols = set()
+    seen_subexp = OrderedSet()
```

Comment:
No need for OrderedSet here as far as I can see -- the sets are used for containment checks and are not iterated over. But double check.